### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by ackleyd1** [ackleyd1/MI-449-SS17-740-js-conditionals-uMDt4P#1]
@chrisvfritz Can you take a look at this? It's [hosted here](https://buddha.surge.sh/) and meets the following criteria:

- [x] Use the `prompt` and `confirm` functions (each at least once) to collect choices from the user (the entire story will take place in dialog boxes, so your `body` element could actually be empty, apart from the `script` element to include your JavaScript)
- [x] Use `if`, `else if`, and `else` to branch the story, based on user choices
- [x] In at least one branch, there should be at least 3 choices for the user to make
- [x] Include at least one branch where different things can happen each time with `Math.random()`
- [x] Use a logical operator (`&&` or `||`) at least once, where it makes sense
- [x] It should be impossible for an error to ever occur when playing your game, including handling for `null`/`undefined` where appropriate
- [x] All string input should be normalized, so that extra whitespace or a letter that's the wrong case will never cause a failed match
- [x] At least one number value is collected and used in a comparison (don't forget to convert it into a number first)
- [x] Don't get _too_ carried away -- it can be tempting to build a complex video game, but remember, you can always come back and build on this project later

I don't have 3 choices or a number, and im definitely getting carried away.
I'm at the Hatch and on the Discovery board of companies theres one called Ear Buddha. I always laughed at it wondering what it could possibly be. 

![ms-qxgiaf](https://cloud.githubusercontent.com/assets/12974323/23016097/5bea79b0-f403-11e6-9c38-ae6e3fe81337.gif)

Perhaps it has to do with Buddha's giant ears.

Anyway it drew my inspiration for this project, and after reading somewhere that the Buddha learned 
64 different languages, I decided to base this project around that. So I'm just asking you to poke through my js, some funky stuff is going on, mainly because of [w3schools outdated google translate guide](https://www.w3schools.com/howto/howto_google_translate.asp). Also there might be other things. 

I want it so that you can enter what you want to learn in chinese/japanese/korean and it will get the answer in a string, and preferably tell you it back in a confirm box. Im doing it right now by changing my div element that they suggested. Once i init the google translate, i cant do any other alerts/confirms/prompts. 

Suggestions?

**Comment by ackleyd1** [comment]
Might also note that it doesnt work on my surge site, yet somewhat works running live-server

**Comment by chrisvfritz** [comment]
Interesting idea! This is a bit tricky, because doing it "the right way" with [the Google Translate API](https://cloud.google.com/translate/docs/translating-text) would require introducing some asynchronous Ajax requests. If you wanted though, you could [sign up here](https://cloud.google.com/translate/) for 2 free months and give it a shot. I'd be happy to walk you through anything you don't completely understand. A video call would probably be useful. 🙂 

A few other items I noticed:

### `trim` and `toLowerCase` are _non_-mutative

This:

``` js
strToNormalize.toLowerCase().trim()
```

Doesn't actually do anything, because these methods merely return an updated _copy_ of the string. The value in `strToNormalize` still has not changed.

### Default parameters dangerous in JavaScript

Referring to:

``` js
function (language = 'en') {
```

Unfortunately, this feature of JavaScript (default parameters) is [not widely supported yet](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters#Browser_compatibility). For example, it doesn't work in any version of Internet Explorer or any version of mobile Safari below 10. For that reason, I'd recommend either avoiding it altogether or [using something like Babel](https://onsen.io/blog/trying-out-javascript-es6-using-babel/) to compile new JS features to well-supported ES5 code. This opens up a whole new can of worms though, so I probably wouldn't recommend it just yet.

**Comment by ackleyd1** [comment]
Hahah I really take a lot of other programming language features for granted and try them in others foolishly. try an else clause after a for loop in c++. not gonna happen. but try it in python and everything is hunky dory.

Also I hated pointers and references when i started learning them in c++ and c (well no references in c but all ya kinda need is a pointer anyway) but now i really like them :( nothing in javascript.

Do you think we can do the ajax stuff tomorrow on a video call? I'd be super interested to dig a bit into those google apis and learn more.

**Comment by chrisvfritz** [comment]
Definitely! Late afternoon or evening would work best for me, if that works for you.

**Comment by ackleyd1** [comment]
@chrisvfritz @egillespie I wanted to get google API's down but I think it was too much for me to handle for now so i reworked the project to meet the requirements as is. One thing is I have some yes or no questions that i just use the confirm function which I thought served its purpose. Is that ok? it is now hosted [here](http://greys.surge.sh/)

**Comment by egillespie** [comment]
You've done a great job with the project. Sorry you weren't able to keep going with Google Translate.

To answer your question about `confirm`, you're using it correctly in some places, but not others. Anywhere that you need to collect a yes/no answer from a user is a good place for `confirm` (like you've done on [line 18](https://github.com/ackleyd1/MI-449-SS17-740-js-conditionals-uMDt4P/blob/master/script.js#L18)).

In cases where you need to show information to a user, but you don't need or care about their input, an `alert` is a better choice. For example, on [line 17](https://github.com/ackleyd1/MI-449-SS17-740-js-conditionals-uMDt4P/blob/master/script.js#L17) you display information but do not save the result of the `confirm` to a variable. This would be a good place for an `alert`.

Update your code to match this paradigm and comment back here once you've pushed your changes to GitHub and Surge. I'll take another look when it's ready.

Thanks! 👽

**Comment by ackleyd1** [comment]
Updated

**Comment by egillespie** [comment]
Preesh! :shipit:
